# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpConnectionMgrChild.cpp
# Commit: 2502449c2e24
# Full Hash: 2502449c2e24993d495c886c717e7f9546cd9e17
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-05 21:44:54
# Regressor Bug: 1497249
# File Overlap Count: 4
# Description:
#   Bug 1497249 - P2: ipdl for nsIWebSocketConnection r=michal
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30624
# ==============================================================================

diff -r 432c7e9dc19f -r 2502449c2e24 netwerk/protocol/http/HttpConnectionMgrChild.cpp
--- a/netwerk/protocol/http/HttpConnectionMgrChild.cpp	Mon Aug 03 10:03:08 2020 +0000
+++ b/netwerk/protocol/http/HttpConnectionMgrChild.cpp	Wed Aug 05 15:17:38 2020 +0000
@@ -11,6 +11,7 @@
 #include "HttpTransactionChild.h"
 #include "AltSvcTransactionChild.h"
 #include "EventTokenBucket.h"
+#include "mozilla/net/WebSocketConnectionChild.h"
 #include "nsHttpConnectionInfo.h"
 #include "nsHttpConnectionMgr.h"
 #include "nsHttpHandler.h"
@@ -177,5 +178,25 @@
   return IPC_OK();
 }
 
+already_AddRefed<PWebSocketConnectionChild>
+HttpConnectionMgrChild::AllocPWebSocketConnectionChild(
+    PHttpTransactionChild* aTransWithStickyConn) {
+  RefPtr<WebSocketConnectionChild> actor = new WebSocketConnectionChild();
+  return actor.forget();
+}
+
+mozilla::ipc::IPCResult
+HttpConnectionMgrChild::RecvPWebSocketConnectionConstructor(
+    PWebSocketConnectionChild* aActor,
+    PHttpTransactionChild* aTransWithStickyConn) {
+  RefPtr<WebSocketConnectionChild> child =
+      static_cast<WebSocketConnectionChild*>(aActor);
+  nsCOMPtr<nsIHttpUpgradeListener> listener =
+      static_cast<nsIHttpUpgradeListener*>(child.get());
+  Unused << mConnMgr->CompleteUpgrade(
+      ToRealHttpTransaction(aTransWithStickyConn), listener);
+  return IPC_OK();
+}
+
 }  // namespace net
 }  // namespace mozilla